docs/core/diagnostics/well-known-event-providers.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Well-known event providers in .NET
+description: Review the providers and events published by the .NET runtime and libraries.
+ms.topic: reference
+ms.date: 12/21/2020
+---
+
+# Well-known event providers in .NET
+
+The .NET runtime and libraries write diagnostic events through a number of different event providers. Depending on your diagnostic needs, you can choose the appropriate providers to enable. This document describes some of the most commonly used event providers in the .NET runtime and libraries.
+
+## CoreCLR
+
+### "Microsoft-Windows-DotNETRuntime" provider
+
+This provider emits various events from the .NET runtime, including the GC, Loader, JIT, Exception and other events from the runtime. Read more about each event from this provider in [Runtime Provider Events List](../../fundamentals/diagnostics/runtime-events.md).
+
+### "Microsoft-DotNETCore-SampleProfiler" provider
+
+This provider is a .NET runtime event provider that is used for CPU sampling for managed callstacks. When enabled, it captures a snapshot of all threads' managed callstacks every 10 ms. To enable this, you need to specify an <xref:System.Diagnostics.Tracing.EventLevel> of `Informational` or higher.
+
+## Framework Libaries
+
+### "Microsoft-Extensions-DependencyInjection" provider
+
+This provider logs information from DependencyInjection. The table below shows events logged by `Microsoft-Extensions-DependencyInjection` provider:
+
+|Event name|Level|Description|
+|----------|-----|-----------|
+|`CallSiteBuilt`|Verbose (5)|A call site has been built.|
+|`ServiceResolved`|Verbose (5)|A service has been resolved.|
+|`ExpressionTreeGenerated`|Verbose (5)|An expression tree has been generated.|
+|`DynamicMethodBuilt`|Verbose (5)|A <xref:System.Reflection.Emit.DynamicMethod> has been built.|
+
+### "System.Buffers.ArrayPoolEventSource" provider
+
+This provider logs information from the ArrayPool. The table below shows the events logged by `ArrayPoolEventSource`:
```

Comment:
Please change throughout. We're not supposed to use "above" and "below" because the layout can differ.

```suggestion
This provider logs information from the ArrayPool. The following table shows the events logged by `ArrayPoolEventSource`:
```